aips/aip-6.md

```diff
@@ -0,0 +1,84 @@
+
+In the current staking implementation, the activeness of a validator is influenced by a single entity's stake (stake-pool owner) which can leave the node unstaked at any time (actually on lockup period expiration). In the new implementation, this scenario is less likely to occur.
+
+## Rationale
+
+This feature has the potential to increase the number of validators in the ecosystem leading to further decentralization and a higher amount of tokens staked on the protocol.
+
+## Specification
+
+A detailed specification of the proposed solution can be found [here](https://docs.google.com/document/d/1wmE_TV3AsYP_lAtSYOZg9tqJZ9_BeRCvfrz7OHXv7zA/edit?usp=sharing).
+
+To summarize, the following behavior would be supported:
+
+- Admin of the delegation pool = node operator
+- Initiates delegation pool
+- Join validator set
+- Leave validator set
+- Set commission rate at the start of contract
+  - Contract will pay commission out first, then rewards to principal
+  - Commission rate cannot be changed
+- Delegators = token owners
+  1.  Add stake
+  2.  Schedule stake for unlocking
+  3.  Cancel unlocking of stake (moves stake to previous state)
+  4.  Withdraw stake
+- Reset-lockup = No one
+- Delegated voter = Admin
+
+## Reference Implementation
+
+There is a reference autonomous (external to the Aptos framework) and tested implementation treating the stake module as a black box at https://github.com/bwarelabs/aptos-delegation-pool.
+Additionally, an example delegation pool, linked to a running validator node and active on the Aptos testnet, can be examined on the [explorer](https://explorer.aptoslabs.com/account/0x61f9e2697d5eee926e990b537a8286a029c908c1ae117e76ddf7a849bf87fb59/resources).
+
+The integration into the Aptos framework would preserve the rewards-distribution formula and the accounting of delegators' stakes, while deprecating resources already available within the framework and responsible for tracking epochs, rewards and pool's aggregated stakes.
+
+The delegation pool would remain a proxy resource for the underlying stake pool, while the factory module could support the wrapping of existing stake pools into delegation ones in the future.
+
+## Risks and Drawbacks
+
+The staking API initially exposed would incur a higher gas cost (only for delegation pools) as additional resources have to be stored and maintained in order to keep track of individual delegators' stakes and cumulative rewards earned by pool's aggregated stakes.
+
+Compared to a single-owner stake pool, rewards produced each epoch would not be automatically restaked for delegators as this would introduce a quadratic complexity when updating the validators set at the epoch's end. Nonetheless, delegators can manually restake their earned rewards, while failing to do so over an entire year would result in their compound APR dropping from 16.18% (automatically restaking each epoch of 1h duration) to 15% (current reward rate).
```

Comment:
"rewards produced each epoch would not be automatically restaked" I'm not sure if I follow this. Rewards produced at the end of each epoch are automatically added to the stake pool so they're restaked by default. Is this referring to this or something else?

Reply:
This applied to the previous version of delegation-pool. Currently, the use of shares-based pool ensures rewards are automatically restaked due to share appreciation